dds/idl/topic_keys.cpp

```diff
@@ -316,8 +322,12 @@ TopicKeys::Iterator::path()
 void
 TopicKeys::Iterator::path_i(std::stringstream& ss)
 {
+  const std::string& error_msg = "Can't get path for invalid topic key iterator!";
```

Comment:
```suggestion
  const std::string error_msg = "Can't get path for invalid topic key iterator!";
```
Or just `const char*` and let it construct a `std::string` on demand